File: src/shared/classes/Image.ts
export class Image {
  url: string | null = null
  height: number | null = null
  width: number | null = null

  static from(json: any){
    if (json === null) return null
    
    let image = new Image()
    
    image.url = json?.url
    image.height = json?.height
    image.width = json?.width
    
    return image
  }
}

File: pkg/expandify/models/embeds/image.go
package embeds

import "github.com/zmb3/spotify/v2"

type Image struct {
	// gorm.Model
	Height int    `json:"height"`
	Width  int    `json:"width"`
	URL    string `json:"url"`
}

func NewImage(image *spotify.Image) Image {
	if image == nil {
		return Image{}
	}

	return Image{
		Height: image.Height,
		Width:  image.Width,
		URL:    image.URL,
	}
}
